src/Components/TitledContainerWithButton/TitledContainerWithButton.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+import {Box, Typography, Button } from '@mui/material';
+
+export  const TitledContainerWithButton=({containerTitle,buttonTitle,children})=> {
+
+  return (  
+    
+    <Box container display={'flex'} flexDirection={'column'} sx={{
+      paddingBlock:"0.2rem",
+      paddingLeft:"1rem",
+      marginBlock:'1rem',
+    }} >
+
+      <Box sx={{
+        display:"flex",marginBlockEnd:".5rem",
+        justifyContent:"space-between",
+        alignItems:"baseline"
+      }}>
+
+        <Typography 
+       
+         sx={{
+                color: "dark.main",
+                fontWeight: "fontWeightLabelSmall",
+                fontSize: { xs: " 0.875rem",sm:"1.2rem", md: "1.4rem" },
+            }}>
+              
+           {containerTitle}
+            
+         </Typography>
+
+        {
+        (buttonTitle)?
+        <Button variant='text'>
+
+          <Typography px={'0.5rem'} sx={{
+            fontWeight:"fontWeightMedium",
+            fontSize: { xs: " 0.875rem",sm:"1rem", md: "1.2rem" },
+
+             
+          }}>{buttonTitle}
+            </Typography>
```

Comment:
The Button component contains another Typography component. This is unnecessary as the Button component itself can contain text. You could just do:

```
<Button variant='text' sx={{ fontWeight:"fontWeightMedium", fontSize: { xs: " 0.875rem",sm:"1rem", md: "1.2rem" } }}>
  {buttonTitle}
</Button>
```

if it didn't work as you want, then keep the typography.